aristanetworks/ansible-avd
# Move cURL validation in mkdocs container + CI simplification

## Change Summary

## Types of changes

- [x] Other (please describe): Github Actions

## Related Issue(s)

Fixes issue #671 

## Component(s) name

`github actions`

## Proposed changes

- Run cURL validation in mkdocs container
- Implement a Makefile target to simplify test & CI

## How to test

```bash
$ make ci-check-links-avd
--------------------------
Starting docker stack for testing
--------------------------
cp development/docker-compose.yml . \
        && sed -i '.original' 's/ansible-avd\///g' docker-compose.yml \
        && docker-compose -f docker-compose.yml up -d webdoc_avd \
        && docker-compose -f docker-compose.yml ps
Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/
Creating network "ansible-avd_default" with the default driver
Creating webdoc_avd ... done
   Name                 Command               State           Ports
----------------------------------------------------------------------------
webdoc_avd   sh -c pip install -r ansib ...   Up      0.0.0.0:8000->8000/tcp

--------------------------
Checking if web server is started
--------------------------
until docker exec webdoc_avd curl -s -I http://localhost:8000/ ; do sleep 2; done
HTTP/1.1 200 OK
Server: TornadoServer/6.1
Content-Type: text/html
Date: Sat, 06 Feb 2021 18:07:39 GMT
Accept-Ranges: bytes
Etag: "9002441ba589444f306d1bf5935b627e"
Last-Modified: Sat, 06 Feb 2021 18:07:28 GMT
Content-Length: 32449


--------------------------
Check HTTP link
--------------------------
docker run --network container:webdoc_avd raviqqe/muffet:1.5.7 http://127.0.0.1:8000/ -e ".*fonts.gstatic.com.*" -e ".*tools.ietf.org.*" -e ".*edit.*" -f --limit-redirections=3 --timeout=30

--------------------------
Shutting down docker stack
--------------------------
docker-compose -f docker-compose.yml down \
        && rm -f docker-compose.*
Stopping webdoc_avd ... done
Removing webdoc_avd ... done
Removing network ansible-avd_default

```

## Checklist:
